pytests/func/test_django.py

```diff
@@ -0,0 +1,190 @@
+
+    assert web_result['content'].find(bp_var_content) != -1
+
+    # shutdown to web server
+    link += 'exit' if link.endswith('/') else '/exit'
+    with get_web_content(link):
+        pass
+
+def test_django_breakpoint_no_multiproc(debug_session):
+    _django_breakpoint_no_multiproc(debug_session, DJANGO1_MANAGE, 40, 'home')
+
+def test_django_template_breakpoint_no_multiproc(debug_session):
+    _django_breakpoint_no_multiproc(debug_session, DJANGO1_TEMPLATE, 8, 'Django Template')
+
+
+def _django_exception_no_multiproc(debug_session, ex_type, ex_line):
+    _django_no_multiproc_common(debug_session)
+    debug_session.prepare_to_run(filename=DJANGO1_MANAGE)
+
+    debug_session.send_request('setExceptionBreakpoints', arguments={
+        'filters': ['raised', 'uncaught'],
+    }).wait_for_response()
+
+    debug_session.start_debugging()
+
+    base_link = _wait_for_django_link(debug_session)
+    assert base_link is not None
+
+    link = base_link + ex_type if base_link.endswith('/') else ('/' + ex_type)
+    with get_web_content(link, {}):
+        thread_stopped = debug_session.wait_for_next(Event('stopped', ANY.dict_with({'reason': 'exception'})))
+        assert thread_stopped == Event('stopped', ANY.dict_with({
+            'reason': 'exception',
+            'text': 'ArithmeticError',
+            'description': 'Hello'
+        }))
+
+        tid = thread_stopped.body['threadId']
+        resp_exception_info = debug_session.send_request(
+            'exceptionInfo',
+            arguments={'threadId': tid, }
+        ).wait_for_response()
+        exception = resp_exception_info.body
+        assert exception == {
+            'exceptionId': 'ArithmeticError',
+            'breakMode': 'always',
+            'description': 'Hello',
+            'details': {
+                'message': 'Hello',
+                'typeName': 'ArithmeticError',
+                'source': DJANGO1_MANAGE,
+                'stackTrace': ANY,
+            }
+        }
+
+        resp_stacktrace = debug_session.send_request('stackTrace', arguments={
+            'threadId': tid,
+        }).wait_for_response()
+        assert resp_stacktrace.body['totalFrames'] > 1
+        frames = resp_stacktrace.body['stackFrames']
+        assert frames[0] == {
+            'id': ANY,
+            'name': 'bad_route_' + ex_type,
+            'source': {
+                'sourceReference': ANY,
+                'path': DJANGO1_MANAGE,
+            },
+            'line': ex_line,
+            'column': 1,
+        }
+
+        debug_session.send_request('continue').wait_for_response()
+        debug_session.wait_for_next(Event('continued'))
+
+    # shutdown to web server
+    link = base_link + 'exit' if base_link.endswith('/') else '/exit'
+    with get_web_content(link):
+        pass
+
+
+def test_django_handled_exception_no_multiproc(debug_session):
+    _django_exception_no_multiproc(debug_session, 'handled', 50)
```

Comment:
For this sort of parametrization, consider using `@pytest.parametrize` instead of helpers. You can parametrize on groups of arguments instead of single ones:
```py
@pytest.mark.parametrize('ex_type, ex_line', [
  ('handled', 50),
  ('unhandled', 64),
])
def test_django_no_multiproc(debug_session, ex_type, ex_line):
   ...
```